### DISCUSSION TITLE
No warning about possilbe null value in tuple

### DISCUSSION THREAD

**Discussion by asv7c2** [dotnet/roslyn#68531] [Q&A] [answered]
```
try
{
    var t = Tuple(); 
    var count = t.Item1.Count; // System.NullReferenceException: Object reference not set to an instance of an object.

    OneValue();
}
catch (Exception e)
{
    Console.WriteLine(e);
}

(List<int>, int) Tuple()
{
    return default; // No warning
}

List<int> OneValue()
{
    return default; // [CS8603] Possible null reference return.
}
```

It is a bug? Must be a warning about possilbe null value in Tuple() return

**Comment by tfenise** [comment, marked as answer]
https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references#structs

`(List<int>, int)` is just the struct `ValueTuple<List<int>, int>`, so the pitfall above applies here.

**Reply by RikkiGibson** [reply]
This pitfall applies here, although one could imagine an enhancement to nullable analysis, specific to tuples, where `return default;` in `Tuple()` is treated similarly to `return (null, 0);`.
